api/signup.go

```diff
@@ -12,6 +12,33 @@ type SignupOrg struct {
 	Subdomain string `json:"subDomain"`
 }
 
+var reservedSubDomains = []string{
+	"infra", "infrahq", "auth", "authz", "authn",
+	"api", "www", "ftp", "ssh", "info", "help", "about",
+	"grants", "connector", "login", "signup",
+	"system", "admin", "email", "bastion",
+}
+
+func (r SignupOrg) ValidationRules() []validate.ValidationRule {
+	return []validate.ValidationRule{
+		validate.Required("name", r.Name),
+		validate.Required("subDomain", r.Subdomain),
+		validate.ReservedStrings("subDomain", r.Subdomain, reservedSubDomains),
+		validate.StringRule{
```

Comment:
not necessarily a blocker, but it's missing the can't-start-with-dash case

Reply:
I added first character validation to the `StringRule` and added it here as well